### DISCUSSION TITLE
Using data with text, start and end time in one array

### DISCUSSION THREAD

**Discussion by PhilippDehner** [marcusolsson/grafana-gantt-panel#41] [Q&A] [answered]
I have data from influx which includes the name (-> text) and a start and end time (see screenshot).
The problem is that it's not possible to select the text option, there's no option given to select. 

Can this be done by an improvement of the panel? (Or would it be better to use another format of the data in the query? -> then we can close this issue and I will use the discussion tab)

![gantt_data](https://user-images.githubusercontent.com/49042626/122030214-df1ad700-cdcd-11eb-9374-8e8120e5683a.png)

**Comment by marcusolsson** [comment]
The data in your screenshot looks right to me at first glance. You need to Format as Table, which you seem to have, and you seem to have the required fields (String, Time, Time).

Could you check the DataFrame structure in the query inspector? I'd like to know the actual type of the fields are. The Dimension drop down filter based on the expected type. If the field isn't listed, that likely means that the type is wrong.

**Comment by PhilippDehner** [comment]
> Could you check the DataFrame structure in the query inspector?

Here's the json, I hope, it's the right one.

[https://gist.github.com/trophi20/68ad2b6cf9a8204cdef3319b147b78f5](https://gist.github.com/trophi20/68ad2b6cf9a8204cdef3319b147b78f5)

**Comment by marcusolsson** [comment, marked as answer]
Thank you! From the JSON you gave me it looks like you have three separate (time series) data frames, each with a Time and Number field. This is not a supported format.

In order for to be able to use the Gantt panel, your data frame needs to have one data frame with three fields of type String, Time, Time. That being said, I'm not sure how you'd need to update your query (I'm not very familiar with InfluxDB). For starters, can you confirm that you've set **Format as** to **Table**?

**Comment by PhilippDehner** [comment]
Yes, that helped me a step further. But now there's the following error:
![grafik](https://user-images.githubusercontent.com/49042626/122053421-03cf7880-cde7-11eb-99d0-39b0d4781ce7.png)

I don't understand why this variable is named "t", because I builded the code not for production, so the variables are actually not shortened. 

`TypeError: t is undefined
    getCalculator fieldColor.ts:38
    x scale.ts:37
    S displayProcessor.ts:104
    GanttChart GanttChart.tsx:321
    GanttChart GanttChart.tsx:299
    React 7
    unstable_runWithPriority scheduler.production.min.js:18
    React 6
    onDataUpdate PanelChrome.tsx:221
    next PanelChrome.tsx:127
    RxJS 19
    useLastResultFrom PanelQueryRunner.ts:314
    d actions.ts:107
    setTimeout handler*d/< actions.ts:106
    Redux 2
    componentWillUnmount PanelEditor.tsx:118
    React 3
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 5
react-dom.production.min.js:216:191
    React 5
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 6
    onDataUpdate PanelChrome.tsx:221
    next PanelChrome.tsx:127
    RxJS 19
    useLastResultFrom PanelQueryRunner.ts:314
    d actions.ts:107
    (Async: setTimeout handler)
    d actions.ts:106
    Redux 2
    componentWillUnmount PanelEditor.tsx:118
    React 3
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 5
`

**Reply by marcusolsson** [reply]
I think you just bumped into https://github.com/marcusolsson/grafana-gantt-panel/issues/38. Could you open the Developer Tools on your machine and check the Console tab for any error/stack traces, and add it to that issue?
